chore: updating deprecated button shape to align with rounded rectangle

georgewrmarshall opened the pull request:
## **Description**

This PR updates old button styling across the application to align with our [brand evolution and home page](https://consensys.slack.com/archives/C0354T27M5M/p1747135800699529) redesign. This PR is a [one line file change](https://github.com/MetaMask/metamask-extension/pull/31818/files?file-filters%5B%5D=.tsx&show-viewed-files=true#diff-6a6f56e439ab2ab776db250936c8bb3f6ae127824f26022e5a9ca61590411c60R89) the rest are all snapshot updates that change the classname.

### Related
- [Slack thread](https://consensys.slack.com/archives/C0354T27M5M/p1747135800699529)
- Equivalent mobile PR https://github.com/MetaMask/metamask-mobile/pull/14540
- Component Library Button PR https://github.com/MetaMask/metamask-extension/pull/31818
- Custom border radius overrides PR https://github.com/MetaMask/metamask-extension/pull/32809

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/32802?quickstart=1)

## **Related issues**

Part of: https://github.com/MetaMask/metamask-extension/issues/31616

## **Manual testing steps**

1. Start storybook `yarn storybook`
2. Navigate to the old button component
3. Verify buttons now display with rounded rectangle shape instead of fully rounded
4. Ensure button behavior remains unchanged and only appearance is affected

## **Screenshots/Recordings**

### **Before**
Buttons with fully rounded shape in storybook

https://github.com/user-attachments/assets/09b056a0-c98e-4a94-9833-0946c7bd779e

### **After**
Buttons with rounded rectangle shape in storybook

https://github.com/user-attachments/assets/d1294d33-089c-455f-8608-7e7b3ff509c3

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

georgewrmarshall committed:
chore: updating deprecated button shape to align with rounded rectangle

georgewrmarshall committed:
chore: removing default args from storybook

georgewrmarshall commented on ui/components/ui/button/button.component.js:
```diff
@@ -39,7 +39,7 @@ const Button = ({
   icon,
   className,
-  rounded = true,
+  rounded = false,
```
Setting `rounded` to `false` now defaults to the rectangle button shape — ironically, the same default we had back in [[circa 2021](https://github.com/MetaMask/metamask-extension/commit/9fa1e4203161a98887c9c753a2f5ce92afda55c0#diff-ea639b753e8c63290dc74049c079c4386f340b3b3f407f0164bf0c9cf0742ecdR35)](https://github.com/MetaMask/metamask-extension/commit/9fa1e4203161a98887c9c753a2f5ce92afda55c0#diff-ea639b753e8c63290dc74049c079c4386f340b3b3f407f0164bf0c9cf0742ecdR35). Looks like history really does repeat itself 😅

NidhiKJha reviewed (commented):
PR looks good to me! But in the updated files, unlock page is also updated. It doesn't seem to be round in the actual UI. It still looks round to me 
![Screenshot 2025-05-13 at 3 37 06 PM](https://github.com/user-attachments/assets/9ee52bcb-21cb-4d51-852e-ceb168323820)

NidhiKJha commented:
Same is happening in storybook for me!

georgewrmarshall commented on ui/components/ui/button/button.stories.js:
```diff
@@ -48,13 +48,6 @@ export default {
     className: { control: 'text' },
     onClick: { action: 'clicked' },
   },
-  args: {
-    disabled: false,
-    large: false,
-    submit: false,
-    className: '',
-    rounded: true,
-  },
```
Uses defaults set in component instead of overriding the story

<img width="1512" alt="Screenshot 2025-05-14 at 8 25 32 AM" src="https://github.com/user-attachments/assets/dd4caf71-1fc6-4445-b5fc-a1634b7d9fac" />

georgewrmarshall commented:
> PR looks good to me! But in the updated files, unlock page is also updated. It doesn't seem to be round in the actual UI. It still looks round to me ![Screenshot 2025-05-13 at 3 37 06 PM](https://private-user-images.githubusercontent.com/39872794/443260392-9ee52bcb-21cb-4d51-852e-ceb168323820.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcxNzczMjAsIm5iZiI6MTc0NzE3NzAyMCwicGF0aCI6Ii8zOTg3Mjc5NC80NDMyNjAzOTItOWVlNTJiY2ItMjFjYi00ZDUxLTg1MmUtY2ViMTY4MzIzODIwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEzVDIyNTcwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwNjMxNjA1MzJlZWZkYjUwOTM1ZTQ2ZDE5ZjkxZjQ1YWE0ZmUwOTVhYTA4OTQxMGE2NTAxNzhmMGM2YjU2NzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.igQKmUL7nfzHsa_Hw1sjnkJjfBbX3foDmMRGt2OVTT4)

Great catches @NidhiKJha thank you for the thorough review 🙏 🦅 👀 You are absolutely right! I forgot to push an update to the Button story to rely on the default props [here](https://github.com/MetaMask/metamask-extension/pull/32802/commits/0c5c5074e2aed3c4cb3e2911dee7f5e8a429bc20). I should have also clarified that any custom overrides to the Button I've moved to a separate PR including the Unlock Page button https://github.com/MetaMask/metamask-extension/pull/32809

georgewrmarshall committed:
chore: border radius update to match component library component

georgewrmarshall commented on ui/components/ui/button/buttons.scss:
```diff
@@ -13,7 +13,7 @@
   align-items: center;
   box-sizing: border-box;
-  border-radius: 6px;
+  border-radius: 12px;
```
Matches `BorderRadius.XL` which is set in component library `Button` border radius update https://github.com/MetaMask/metamask-extension/pull/31818/files?file-filters%5B%5D=.tsx&show-viewed-files=true#diff-6a6f56e439ab2ab776db250936c8bb3f6ae127824f26022e5a9ca61590411c60R89

amandaye0h reviewed (approved):
Looks great! Excited to get this in :)
